docs/configuration/intra_node_data_path.md

```diff
@@ -0,0 +1,170 @@
+
+![Example of use_latest_message](../imgs/configuration_use_latest_message.png)
+
+In the figure, latest messages of `ping` topic is mapped to messages of `/pong` topic mechanically by CARET. It is not still easy to understand with only the block figure, but the provided timing chart below help you to understand what `use_latest_message` is.
+
+![Timing chart of use_latest_message](../imgs/timing_chart_use_latest_message.png)
+
+In the timing chart, colored boxes represent duration of callback running, and bold lines are message flow of topic message. Dotted lines indicates mapping between input messages and outputs. With `use_latest_message`, CARET assumes that output message is made from the most recent input message. `use_latest_message` is fairly simple and works well in most cases.
+
+In the timing chart, red dotted lines explains a pitfall of `use_latest_message`. Using `use_latest_message` policy, CARET believes that input message, which is not even processed completely, is mapped to output message. For example, CARET interprets that `/pong[4]` is made from `/ping[2]` because `/ping[2]` is the recent input message for `/pong[4]`. However, `subscription_callback_0` is processing `/ping[2]` and does not share it with `timer_callback_1` before publishing `/pong[4]`. You have to tell CARET the node structure when you find such pitfall.
+
+### `callback_chain`
+
+`callback_chain` is introduced for CARET to map input messages to outputs based on inter-operation of multiple callback functions. Input messages are consumed in subscription callbacks and propagated to other nodes. It looks as if input messages passes chains of multiple callbacks to make output messages. With `callback_chain`, CARET take care of input propagation on callbacks and it is helpful to escape the limitation of `use_latest_message` as mentioned above.
+
+Next figure shows how CARET interprets intra-node data path using `callback_chain`. Intra-communication between `subscription_callback_0` and `timer_callback_1` is taken into account for defining intra-node data path. `variable_passings` a tag used in CARET, and shows such intra-communication.
+
+![Example of callback_chain](../imgs/configuration_callback_chain.png)
+
+The following timing chart shows how input messages are mapped to output messages.
+
+![Timing chart of callback_chain](../imgs/timing_chart_callback_chain.png)
+
+CARET maps mechanically messages of `/pong` topic to messages of `/ping` which finish being processed on `subscription_callback_0`. The unexpected behavior of `use_latest_message` is improved by `callback_chain`.
+
+`callback_chain` looks the best choice. However, it has several drawbacks.
+
+- It is not designed for node which have multiple callbacks running at parallel, and response time might be longer than actual
+- It is not able to detect actual time where buffered data are consumed because it does not trace events in application
```

Comment:
How about using `user code`? I'd like to replace this sentence by the following one.

```
- It is not able to detect actual time when buffered data are consumed because only CARET does not trace user code on ROS 2
```

Reply:
@takeshi-iwanari

Reply:
@takeshi-iwanari fixed in https://github.com/tier4/CARET_doc/commit/335134c51e36648b6b347860454b56bdfaa5d8fb
Please feel free to your concern if you have another comment.

Reply:
LGTM